lib/puppet/util/selinux.rb

```diff
@@ -137,6 +142,33 @@ def set_selinux_default_context(file)
     nil
   end
 
+  ##
+  # normalize_selinux_category is an internal, method that converts
+  # all selinux categories to the human-readable representation,
+  # avoiding potential issues when mcstransd is not functional.
+  #
+  # It is not marked private because it is needed by File's
+  # selcontext.rb, but it is not intended for use outside of Puppet's
+  # code.
+  #
+  # @return [String] the human-readable category name
+  def normalize_selinux_category(category)
+    # We don't cache this, but there's already a ton of duplicate work
+    # in the selinux handling code.
+
+    File.open(Selinux.selinux_translations_path).each do |line|
+      stripped = line.strip!
+      next if stripped[0] == "#" # skip comments
+      line.gsub!(/\s+/m, '')
```

Comment:
Can we use `[[:space:]]` instead of `\s` ? Helps guard against encoding issues.

> POSIX bracket expressions are also similar to character classes. They provide a portable alternative to the above, with the added benefit that they encompass non-ASCII characters. For instance, /\d/ matches only the ASCII decimal digits (0-9); whereas /[[:digit:]]/ matches any character in the Unicode Nd category.
https://ruby-doc.org/core-2.2.0/Regexp.html. PUP-6875 is related.